sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/ZetaSqlUtils.java

```diff
@@ -158,6 +158,9 @@ public static Object zetaSqlValueToJavaObject(Value value, FieldType fieldType)
       case DECIMAL:
         return value.getNumericValue();
       case DOUBLE:
+        if (value.getType().getKind().equals(TypeKind.TYPE_INT64)) {
```

Comment:
This seems like a symptom of a bug elsewhere in Beam ZetaSQL. Can you add a comment to the code explaining why this is required? (The line from the PR description will be good.)

Reply:
Added a comment.